docs/common/java-install.md

```diff
@@ -0,0 +1,154 @@
+---
+title: Installing or Updating Java
+slug: /java-install-update
+description: How to install or update to Java 17 on Linux (apt/rpm), Windows, or Mac.
+toc_max_heading_level: 5
+---
+
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+Installing Java is a critical first step to using or developing plugins for Paper, Velocity, and
+Waterfall. This guide will walk you through the recommended installation steps for most major
+platforms.
+
+:::tip
+
+This guide focuses on Amazon's Corretto OpenJDK distribution. This is because it offers the best
+installation experience on the most platforms. Corretto is, however, not the only OpenJDK vendor to
+choose from. Many alternatives exist such as [Eclipse Adoptium](https://adoptium.net/),
+[Microsoft](https://www.microsoft.com/openjdk) and
+[Azul Zulu](https://www.azul.com/downloads/?package=jdk). Note that the JDK Oracle distributes,
+while functionally identical, is **not** recommended due to an extremely unfriendly installer and
+previous hostile licensing.
+
+:::
+
+## Linux
+
+### Ubuntu/Debian
+
+Installing Java 17 on Debian-based Linux distributions is very simple. First, ensure your system has
+all required tools to successfully install Java.
+
+```bash
+sudo apt-get update && sudo apt-get upgrade
+sudo apt-get install software-properties-common ca-certificates apt-transport-https curl
+```
+
+Second, import the Amazon Corretto public key and apt repository.
+
+```bash
+curl https://apt.corretto.aws/corretto.key | sudo apt-key add -
+sudo add-apt-repository 'deb https://apt.corretto.aws stable main'
+```
+
+Then, install Java 17.
+
+```bash
+sudo apt-get update
+sudo apt-get install -y java-17-amazon-corretto-jdk
+```
+
+Proceed to [verify your installation](#verifying-installation).
+
+### RPM-based
+
+To install Java 17 on CentOS, RHEL, Fedora, openSUSE, SLES, or any other RPM-based Linux
+distribution, execute the following commands depending on your package manager. Once you have
+finished, precede to [verify your installation](#verifying-installation).
+
+#### DNF
+
+DNF is used on Fedora, CentOS/RHEL 7+, and related distributions.
+
+```bash
+sudo rpm --import https://yum.corretto.aws/corretto.key
+sudo curl -Lo /etc/yum.repos.d/corretto.repo https://yum.corretto.aws/corretto.repo
+sudo dnf -y install java-17-amazon-corretto-devel
+```
+
+#### Zypper
+
+Zypper is used on openSUSE, SLES, and related distributions.
+
+```bash
+sudo zypper addrepo https://yum.corretto.aws/corretto.repo
+sudo zypper refresh
+sudo zypper install java-17-amazon-corretto-devel
+```
+
+#### YUM
+
+YUM is used on older releases of CentOS/RHEL, and excessively old releases of Fedora.
+
+```bash
+sudo rpm --import https://yum.corretto.aws/corretto.key
+sudo curl -Lo /etc/yum.repos.d/corretto.repo https://yum.corretto.aws/corretto.repo
+sudo yum -y install java-17-amazon-corretto-devel
+```
+
+## Windows 10 & 11
```

Comment:
maybe mention winget here, the windows package manager (nobody really cares about I guess but its the fastest way):
`winget install -e --id Amazon.Corretto.17`

Reply:
Is this included out of the box? The docs Microsoft provides are a bit unclear as they provide installation instructions, which say that it is included in a default package, but then show you how to install it through the MS store.

On my windows computer, it wasn't, although admittedly it is through work so it's possible their image excludes it/provides some super outdated version not including it.

Reply:
uuuhm, I don't remember installing it but this install is really old sooooo